### PR TITLE
Evitar que un M2 d'un altre autoconsum active fluxos de GURB

### PR DESCRIPTION
**Pull request by oriolpiera**
Tanca #1250

## Resum
- afegeix un guard de CAU abans d'executar efectes laterals específics de GURB dins de `GiscedataSwitchingM2_05.create_from_xml()`
- actualitza els tests existents d'M2 perquè els casos positius utilitzen el CAU real del GURB
- afegeix un test de regressió per verificar que un CAU d'autoconsum alié no dispara el flux d'activació de GURB

## Validacions
- `python -m py_compile som_gurb/models/giscedata_switching.py som_gurb/tests/tests_gurb_switching.py`
- `dodestral test_som_gurb -m som_gurb` s'ha llançat amb `pyenv activate erp`, però no ha acabat dins del temps disponible perquè l'entorn estava reconstruint una base ERP de proves des de zero

## Notes
- esta PR corregeix la causa arrel verificada en `M2_05`: reaccionar a qualsevol CAU d'autoconsum d'una pòlissa que tinga GURB
- el símptoma funcional de `Baixa pendent` no s'ha reproduït completament dins d'una execució de tests tancada en esta sessió

### CODE REVIEW

File: som_gurb/__terp__.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
         "giscedata_telemesures_base",
         "giscedata_facturacio_services",
         "giscedata_polissa_category",
+        "report_banner_poweremail",
```
Comment (destanyol) [inline som_gurb/__terp__.py]: wtf

File: som_gurb/models/giscedata_switching.py [hunk 2]
```diff
@@ -58,6 +58,18 @@ def _cups_contract_has_gurb_cups(cursor, uid, pool, pol_id, context=None):
...
+    gurb_cau = getattr(gurb_cups.gurb_cau_id.self_consumption_id, "cau", False)
+    if not gurb_cau:
+        return False
+
+    for dades_cau in getattr(step, "dades_cau", []):
+        if dades_cau.cau == gurb_cau:
+            return True
+
+    return False
+
```
Comment (destanyol) [inline som_gurb/models/giscedata_switching.py]: No acabo d'entendre aquesta funció. I pq està fora?

File: som_gurb/tests/tests_gurb_switching.py [hunk 3]
```diff
@@ -1466,6 +1467,10 @@ def test_create_from_xml_m2_05_unexpected_leaving_gurb(
             "<Motivo>01",
             "<Motivo>{0}".format("06")
         )
+        m2_05_xml = m2_05_xml.replace(
+            "<CAU>ES1234000000000001JN0FA001",
+            "<CAU>{0}".format(gurb_cau)
+        )
```
Comment (destanyol) [inline som_gurb/tests/tests_gurb_switching.py:1473]: No acabo d'entendre pq s'han de modificar els testos

**Review by destanyol — CHANGES_REQUESTED** [review]

File: som_gurb/__terp__.py (hunk 1 above)
Comment (oriolpiera) [reply som_gurb/__terp__.py]: em va dir d'afegir això perquè som_gurb o mòduls dels que depen som_gurb, tenien plantilles amb banners !?!

File: som_gurb/models/giscedata_switching.py (hunk 2 above)
Comment (oriolpiera) [reply som_gurb/models/giscedata_switching.py]: mira que el CAU del ATR sigui igual que el CAU del gurb cups
entenc que la posa fora perquè n'ha vist d'altres en aquesta classe (donde vayas)

File: som_gurb/tests/tests_gurb_switching.py (hunk 3 above)
Comment (oriolpiera) [reply som_gurb/tests/tests_gurb_switching.py:1473]: perquè entenc que comprova que marxa o no marxa de gurb, fent que el CAU que està al GURB coincideixi o no amb el del cas ATR que estàs tractant. és un corner case que no estava controloat

File: som_gurb/models/giscedata_switching.py (hunk 2 above)
Comment (oriolpiera) [reply som_gurb/models/giscedata_switching.py]: @destanyol he canviat lo del getattr però lo de fer-ho dins no ho he vist clar, quedava una condició doblement negada bastant difícil d'entendre i ho he vist més clar amb una funció, però l'he passat a dins la classe.

🤺  human generated code

File: som_gurb/__terp__.py (hunk 1 above)
Comment (oriolpiera) [reply som_gurb/__terp__.py]: @destanyol 
mira, acabo d'executar un destral amb som_gurb sense aquesta dependència i surt aquest error:

```
ValidateException: warning -- ValidateError
The value "poweremail.templates,12" for the field "object_ref" is not in the selection
```

El problema és cert que no és de `som_gurb` sinó de `account_invoice_base` que té una plantilla amb banners `banner_domain_env_account_invoice_via_email`, o sigui que l'hauria d'afegir allà la dependència (fer una PR a GISCE, aplicar-la a la intervenció, a la llarga aconseguir que la mergejin, etc..) però ya tu sabe

**Review by destanyol — APPROVED** [review]
